File: 148-Sort-List.js
/**
 * Definition for singly-linked list.
 */
function ListNode(val, next) {
    this.val = (val === undefined ? 0 : val);
    this.next = (next === undefined ? null : next);
}

/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var sortList = function (head) {
    let curr = head,
        values = [];

    while (curr !== null) {
        values.push(curr.val);
        curr = curr.next;
    }

    values.sort((a, b) => b - a);

    let sortedLL = new ListNode(0);
    let dummy = sortedLL;

    while (values.length !== 0) {
        let x = values.pop();
        sortedLL.next = new ListNode(x);
        sortedLL = sortedLL.next;
    }

    return dummy.next;
};

File: 148-Sort-List.py
class Solution:
    def sortList(self, head: Optional[ListNode]) -> Optional[ListNode]:
        dummy = ListNode(0, head)
        nums = []
        curr = head

        while curr:
            nums.append(curr.val)
            curr = curr.next

        nums.sort()
        curr = dummy

        for num in nums:
            curr.next = ListNode(num)
            curr = curr.next

        return dummy.next        